### PR TITLE
fix: Respect `student_view_multi_device` key for HTML xBlocks in the Block API

### PR DESCRIPTION
**Pull request by shafqat-muneer**
[LEARNER-10217](https://2u-internal.atlassian.net/browse/LEARNER-10217): Respect `student_view_multi_device` key for HTML xBlocks in the Block API

For HTML xBlocks, if the `student_view_multi_device` key is set to `false`, the HTML xBlock will not be displayed. Instead, a message will appear stating, "This interactive component isn't available on mobile."

Previous | After Fix |
--- | --- |
<video src="https://github.com/user-attachments/assets/87554cc1-cfd5-45eb-86be-51ec26392e7f" /> | <video src="https://github.com/user-attachments/assets/038adbd7-88cb-46ab-8f93-150a3e2102f0" /> |

### CODE REVIEW

File: Course/Course/Data/CourseRepository.swift
Comment (rnr) [inline Course/Course/Data/CourseRepository.swift]: @shafqat-muneer Do you think we need to override the block type? I can imagine a situation where we would need to know the original type (e.g. we want to have different icons for .html and .unknown types).
Perhaps we could add changes to LessonType like this:
Screenshot 2024-09-16 at 15 25 09
WDYT?
Thank you

Comment (shafqat-muneer) [reply Course/Course/Data/CourseRepository.swift]: @rnr, after a bit of analysis, I agree that it's more appropriate to handle it here. Although we don't have such a case at the moment, there's a chance it could occur in the future.

I've made the update. Thanks for pointing it out!

**Comment by rnr** [comment]
Looks good to me! I know there is a discussion going on about what we should do here, and I will approve this PR once we get approval from Khanan. Thanks

**Comment by miankhalid** [comment]
hi @rnr, 

@touchapp and I discussed this matter on Monday and agreed that this PR is good to go and should be reviewed and approved as necessary.

About the issue with the flag being returned as false, our backend team will look into this issue asap and share their findings with the team very soon. thanks!

**Review by saeedbashir — APPROVED** [review]
LGTM 👍

**Review by rnr — APPROVED** [review]

**Pull request merged**
